pkg/machine/e2e/rm_test.go

```diff
@@ -43,6 +43,49 @@ var _ = Describe("podman machine rm", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(removeSession2).To(Exit(125))
 		Expect(removeSession2.errorToString()).To(ContainSubstring(fmt.Sprintf("%s: VM does not exist", name)))
+
+		// Ensure that the system connections have the right rootfulness
+		name = randomString()
+		i = new(initMachine)
+		session, err = mb.setName(name).setCmd(i.withImage(mb.imagePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		name2 := randomString()
+		i = new(initMachine)
+		session, err = mb.setName(name2).setCmd(i.withImage(mb.imagePath).withRootful(true)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		bm := basicMachine{}
+		sysConnOutput, err := mb.setCmd(bm.withPodmanCommand([]string{"system", "connection", "list", "--format", "'{{.Name}}'"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		names := sysConnOutput.outputToStringSlice()
+		sysConnOutput, err = mb.setCmd(bm.withPodmanCommand([]string{"system", "connection", "list", "--format", "'{{.Default}}'"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		defaults := sysConnOutput.outputToStringSlice()
+		for idx, n := range names {
+			if n == name {
+				Expect(defaults[idx]).To(Equal("true"))
+			}
+		}
```

Comment:
I have not tried to run it but I cannot see how this can work, and if it if it is this is super risk as it can be easily be a NOP. `"'{{.Name}}'"` is not correct usage because the single quote will be part of the template so it will be in the output.

As such you names loop will never match...  

What you should do is something like `"--format", "{{.Name}}--{{.Default}}"`
And then you just match `Expect(sysConnOutput.outputToString()).To(ContainSubstring(name+":true"))`

Then below you do the same match again.

Reply:
That's much cleaner, thank you. I definitely made that more complicated than it needed to be.